tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -46,30 +46,18 @@ public static ObjType DecodeArgsType(DDWAF_OBJ_TYPE t)
             }
         }
 
-        public static ReturnCode DecodeReturnCode(DDWAF_RET_CODE rc)
+        public static ReturnCode DecodeReturnCode(DDWAF_RET_CODE rc) => rc switch
         {
-            switch (rc)
-            {
-                case DDWAF_RET_CODE.DDWAF_ERR_INTERNAL:
-                    return ReturnCode.ErrorInternal;
-                case DDWAF_RET_CODE.DDWAF_ERR_TIMEOUT:
-                    return ReturnCode.ErrorTimeout;
-                case DDWAF_RET_CODE.DDWAF_ERR_INVALID_ARGUMENT:
-                    return ReturnCode.ErrorInvalidArgument;
-                case DDWAF_RET_CODE.DDWAF_ERR_INVALID_OBJECT:
-                    return ReturnCode.ErrorInvalidObject;
-                case DDWAF_RET_CODE.DDWAF_GOOD:
-                    return ReturnCode.Good;
-                case DDWAF_RET_CODE.DDWAF_MONITOR:
-                    return ReturnCode.Monitor;
-                case DDWAF_RET_CODE.DDWAF_BLOCK:
-                    return ReturnCode.Block;
-                default:
-                    throw new Exception($"Unknown return code: {rc}");
-            }
-        }
+            DDWAF_RET_CODE.DDWAF_ERR_INTERNAL => ReturnCode.ErrorInternal,
```

Comment:
`DDWAF_ERR_TIMEOUT` was removed, just confirming that's correct? (Looks like it is).

If so, I'm wondering if there's any danger of version-conflict issues here? Can we be using an older version of the native waf (which includes the old -4 return code) with a _newer_ version of the Tracer library (which throws when it receives -4?) 

> Seems like a problem in the native code to be making breaking changes like that IMO, but seems risk to throw here?

Reply:
In typical scenarios, an older version of the waf works with newer code. We used to get that -4 code when the timeout passed was 0, and when the result internally was DDWAF_GOOD and the timeout was hit, so the first case wouldn't happen and the latter would be really rare.. But it's a good point, and we should be careful about that in the future, but I think it's not worth pushing it further for now, wdyt?